File: src/main/java/dev/kaato/pluginscrud/Java/services/PluginService.java
package dev.kaato.pluginscrud.Java.services;

import dev.kaato.pluginscrud.Java.entities.Plugin;
import dev.kaato.pluginscrud.Java.repositories.PluginRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class PluginService {
    @Autowired
    private PluginRepository repo;

    public List<Plugin> findAll() {
        return repo.findAll();
    }

    public Plugin findById(long id) {
        Optional<Plugin> obj = repo.findById(id);
        return obj.get();
    }

    public Plugin insert(Plugin plugin) {
        return repo.save(plugin);
    }

    public void delete(long id) {
        try {
            repo.deleteById(id);
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }

    public Plugin update(long id, Plugin newPlugin) {
        try {
            Plugin plugin = repo.getReferenceById(id);
            updateData(plugin, newPlugin);
            return repo.save(plugin);
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }

    private void updateData(Plugin plugin, Plugin newPlugin) {
        plugin.setName(newPlugin.getName());
        plugin.setDescription(newPlugin.getDescription());
        plugin.setBanner(newPlugin.getBanner());
        plugin.setLogo(newPlugin.getLogo());
        plugin.setVideo(newPlugin.getVideo());
        plugin.setTopics(newPlugin.getTopics());
        plugin.setStage(newPlugin.getStage());
    }
}

File: src/main/java/dev/kaato/pluginscrud/kotlin/services/PluginService.kt
package dev.kaato.pluginscrud.kotlin.services

import dev.kaato.pluginscrud.kotlin.entities.Plugin
import dev.kaato.pluginscrud.kotlin.repositories.PluginRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import java.util.Optional

@Service
class PluginService @Autowired constructor(val repo: PluginRepository) {
    fun findAll(): List<Plugin?> {
        return repo.findAll()
    }

    fun findById(id: Long): Optional<Plugin?> {
        return repo.findById(id)
    }

    fun insert(plugin: Plugin): Plugin {
        return repo.save(plugin)
    }

    fun delete(id: Long) {
        try {
            return repo.deleteById(id)
        } catch (e: Exception) {
            throw RuntimeException(e)
        }
    }

    fun update(id: Long, newPlugin: Plugin): Plugin {
        try {
            val plugin = repo.getReferenceById(id)
            updateData(plugin, newPlugin)
            return repo.save(plugin)
        } catch (e: Exception) {
            throw RuntimeException(e)
        }
    }

    fun updateData(plugin: Plugin, newPlugin: Plugin) {
        plugin.name = newPlugin.name
        plugin.description = newPlugin.description
        plugin.topics = newPlugin.topics
        plugin.banner = newPlugin.banner
        plugin.logo = newPlugin.logo
        plugin.video = newPlugin.video
        plugin.setStage(newPlugin.getStage())
    }
}